menas/src/main/scala/za/co/absa/enceladus/menas/services/MigrationService.scala

```diff
@@ -15,42 +15,39 @@
 
 package za.co.absa.enceladus.menas.services
 
-import java.io.File
-import java.time.LocalDateTime
-import java.time.format.DateTimeFormatter
-
 import javax.annotation.PostConstruct
-import org.mongodb.scala.MongoDatabase
-import org.mongodb.scala.bson.collection.immutable.Document
+import org.apache.hadoop.conf.Configuration
+import org.apache.log4j.{LogManager, Logger}
+import org.mongodb.scala.{MongoClient, MongoDatabase}
 import org.springframework.beans.factory.annotation.{Autowired, Value}
 import org.springframework.stereotype.Component
-import za.co.absa.enceladus.migrations.MongoMigrator
-import za.co.absa.enceladus.migrations.models.{BackupConfiguration, Evolution, MigratorConfiguration}
-import org.apache.hadoop.conf.Configuration
+import za.co.absa.enceladus.migrations.framework.Migrator
+import za.co.absa.enceladus.migrations.framework.dao.MongoDb
+import za.co.absa.enceladus.migrations.migrations._
+import za.co.absa.enceladus.model._
 
 @Component
 class MigrationService @Autowired()(mongoDb: MongoDatabase, hadoopConf: Configuration) {
 
-  @Value("${za.co.absa.enceladus.migrations.dump.filepath}")
-  private val dumpFilepath: String = ""
   @Value("${za.co.absa.enceladus.menas.mongo.connection.string}")
   val connectionString: String = ""
   @Value("${za.co.absa.enceladus.menas.mongo.connection.database}")
   val database: String = ""
 
-  @PostConstruct
-  def init() = {
-    val fileName = LocalDateTime.now().format(DateTimeFormatter.ofPattern("YYYY-MM-dd-HH-mm-ss"))
-    val backupConf = BackupConfiguration(s"$dumpFilepath${File.separator}$fileName.archive", connectionString, database, Some(hadoopConf))
-    val migratorConf = MigratorConfiguration(backupConf, mongoDb, 10)
-    val migrator = new MongoMigrator(migratorConf)
+  private val log: Logger = LogManager.getLogger(this.getClass)
 
-    migrator.registerOne(Evolution(1, "Create index on evolutions collection", "chochovg").setEvolution { () =>
-      mongoDb.getCollection("evolutions").createIndex(Document("""{"order":1}"""))
-    })
-
-    // Register all evolutions here as above
-    migrator.runEvolutions()
+  @PostConstruct
+  def init(): Unit = {
+    val mongoClient = MongoClient(connectionString)
+    val db = new MongoDb(mongoClient.getDatabase(database))
+
+    if (db.isCollectionExists("schema") || db.isCollectionExists("db_version")) {
```

Comment:
Yeah, the intention for this check is to distinguish between an empty database and an old database (that doesn't have `db_version` collection).

- If there is no `schema` and `db_version` => the database is clearly empty, no migrations needed
- If there is `schema` => the database is not empty
- If there is `db_version` => the database is not empty. The database might not have `schema`, but `schema_v21`. But in this case it should have `db_version` as well.

But I need to improve this part of the code so that all the checks make sense. I might create a function that checks and initializes an empty database.